lib/resty/rediscluster.lua

```diff
@@ -257,6 +325,23 @@ function _M.refresh_slots(self)
     end
 
     self:fetch_slots()
+    -- Cleanup health dict entries for removed nodes
+    local current_nodes = {}
+    local servers = slot_cache[self.config.name .. "serv_list"].serv_list
+    for _, node in ipairs(servers) do
+        current_nodes[node.ip .. ":" .. node.port] = true
+    end
+
+    local health_dict = ngx.shared[DEFAULT_HEALTH_DICT_NAME]
```

Comment:
this shared dict is a global storage, but `serv_list` only for one redis cluster, you can't remove nodes in shared dict by only one cluster's `serv_list`